scipy/sparse/data.py

```diff
@@ -44,6 +44,9 @@ def _imag(self):
         return self._with_data(self.data.imag)
 
     def __neg__(self):
+        if self.dtype.kind == 'b':
+            raise NotImplementedError('inverting a sparse boolean '
+                                      'matrix is not supported')
```

Comment:
This isn't really a constructive comment, but if I saw a traceback with this message then I'd assume matrix inversion is somehow involved (rather than additive inversion). Then again, maybe I'm just thrown off by numpy boolean semantics.
```python
>>> bool(-True)
True
>>> bool(-array(True))
False
```